crates/services/consensus_module/poa/src/service.rs

```diff
@@ -457,13 +491,18 @@ where
 }
 
 #[async_trait::async_trait]
-impl<D, T, B, I> RunnableTask for Task<T, B, I>
+impl<D, T, B, I> RunnableTask for MainTask<T, B, I>
 where
     T: TransactionPool,
     B: BlockProducer<Database = D>,
     I: BlockImporter<Database = D>,
 {
     async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
+        // make sure we're synced first
+        if *self.sync_state.borrow() == SyncState::NotSynced {
+            self.sync_state.changed().await?;
```

Comment:
It is better to handle `watcher` here too, maybe `sync_task` will take a lot of time to work=)